webrender/src/renderer.rs

```diff
@@ -2860,6 +2938,16 @@ pub trait ExternalImageHandler {
     fn unlock(&mut self, key: ExternalImageId, channel_index: u8);
 }
 
+/// Allows callers to receive a texture with the contents of a specific
+/// pipeline copied to it. Lock should return the native texture handle
+/// and the size of the texture. Unlock will only be called if the lock()
+/// call succeeds, when WR has issued the GL commands to copy the output
+/// to the texture handle.
+pub trait OutputImageHandler {
+    fn lock(&mut self, pipeline_id: PipelineId) -> Option<(u32, DeviceIntSize)>;
```

Comment:
we might want to revisit the lock/unlock semantics here later (is that how VR team wants to work with WR?), especially since WR pretty much guarantees that `unlock` is going to be called right away after some blitting (contrary to external image locks, where unlocking can happen at any time later).

Reply:
The vr team have a demo working with the lock/unlock - but yes, we may want to revisit these semantics later on once we work out the exact synchronization story.